lib/services/access_list.go

```diff
@@ -206,6 +213,34 @@ func IsMember(identity tlsca.Identity, clock clockwork.Clock, accessList *access
 	return trace.NotFound("user %s is not a member of the access list", username)
 }
 
+// IsAccessListMember will return true if the user is a member for the current list.
+func IsAccessListMember(ctx context.Context, identity tlsca.Identity, clock clockwork.Clock, accessList *accesslist.AccessList, memberGetter AccessListMembersGetter) error {
+	username := identity.Username
+
+	member, err := memberGetter.GetAccessListMember(ctx, accessList.GetName(), username)
+	if trace.IsNotFound(err) {
+		// The member has not been found, so we know they're not a member of this list.
+		return trace.NotFound("user %s is not a member of the access list", username)
+	} else if err != nil {
+		// Some other error has occurred
+		return trace.Wrap(err)
+	}
+
+	expires := member.Spec.Expires
```

Comment:
Is it possible to have expiration `member.Spec.Expires` after the next audit date? 
If yes, should we consider the expiration as the next audit date?

Reply:
I think that's a good idea. I'll do a minimum of expires vs. spec expiration here.